app/main.py

```diff
@@ -1,2 +1,53 @@
+from dataclasses import dataclass
+from typing import Any, Hashable, Optional
+
+
 class Dictionary:
-    pass
+    @dataclass
+    class Node:
+        key: Hashable
+        value: Any
+
+    def __init__(self, capacity: int = 8) -> None:
```

Comment:
Parametrizing internal structure of a dict is not a very good design choice. If you allow devs to play with dict capacity it will behave differently in different parts of program. And also in real dictionary init method takes key-value pairs that will be stored in dictionary on creation